tst/testinstall/methwhy.tst

```diff
@@ -35,52 +37,90 @@ function ( x )
 end
 gap> Display(ApplicableMethod(foobar, [ ['a'] ], 1));
 #I  Searching Method for foobar with 1 arguments:
-#I  Total: 4 entries
-#I  Method 1: ``foobar: for a string'' at stream:1 , value: 2*SUM_FLAGS+1
+#I  Total: 4 entries, of which 3 are applicable:
+#I  Method 1, valid operation number 1, value: 2*SUM_FLAGS+1
+#I  ``foobar: for a string''
+#I   at stream:1
 function ( x )
     return StartsWith( x, "foo" );
 end
-gap> Display(ApplicableMethod(foobar, [ ['a'] ], 1, 2));
+gap> ApplicableMethod(foobar, 'a', 1);   
+#I  argument 2 must be a list of arguments for the operation
+#I  usage: ApplicableMethod(<opr>,<arglist>[,<verbosity>[,<nr>]])
+fail
+gap> ApplicableMethod(foobar, [ ['a'] ], 1, 3);
+#I  Searching Method for foobar with 1 arguments:
+#I  Total: 4 entries, of which 3 are applicable:
+#I  Method 4, valid operation number 3, value: 0
+#I  ``foobar''
+#I   at stream:1
+function( x ) ... end
+gap> ApplicableMethod(foobar, [ ['a'] ], 1, 4);
 #I  Searching Method for foobar with 1 arguments:
-#I  Total: 4 entries
-#I  Method 1: ``foobar: for a string'' at stream:1 , value: 2*SUM_FLAGS+1
-#I  Skipped:
-#I  Method 2: ``foobar: for a list'' at stream:1 , value: 42
-<Attribute "Length">
+#I  Total: 4 entries, of which 3 are applicable:
+#I  there are only 3 valid methods
+fail
+gap> ApplicableMethod(foobar, [ ['a'] ], 1, "all");
+#I  Searching Method for foobar with 1 arguments:
+#I  Total: 4 entries, of which 3 are applicable:
+#I  Method 1, valid operation number 1, value: 2*SUM_FLAGS+1
+#I  ``foobar: for a string''
+#I   at stream:1
+#I  Method 2, valid operation number 2, value: 42
+#I  ``foobar: for a list''
+#I   at stream:1
+#I  Method 4, valid operation number 3, value: 0
+#I  ``foobar''
+#I   at stream:1
+[ function( x ) ... end, <Attribute "Length">, function( x ) ... end ]
+
+#
 gap> Display(ApplicableMethod(foobar, [fail], 1));
 #I  Searching Method for foobar with 1 arguments:
-#I  Total: 4 entries
-#I  Method 4: ``foobar'' at stream:1 , value: 0
+#I  Total: 4 entries, of which 1 is applicable:
+#I  Method 4, valid operation number 1, value: 0
+#I  ``foobar''
+#I   at stream:1
 function ( x )
     return fail;
 end
-gap> Display(ApplicableMethod(foobar, [fail], 4));
```

Comment:
Why this change?

Reply:
The previous example: Display(ApplicableMethod(foobar, [fail], 1));  displays the function, so there seemed no need to display it again, and: ApplicableMethod(foobar, [fail], 4); gives the shortened version "function( x ) ... end".

Reply:
But the point was precisely to test that `..., 4)` works as intended. That's why this test was there. You seem to be saying we don't need to test it, because we know that it works -- but then we could remove most tests with the same argument, couldn't we?

Reply:
The removed(changed) examples have now been reinstated, and others added.